huaweicloud/common/errors.go

```diff
@@ -199,3 +203,123 @@ func ConvertExpected500ErrInto404Err(err error, errCodeKey string, specErrCodes
 	log.Printf("[WARN] Unable to recognize expected error code (%v), want %v", errCode, specErrCodes)
 	return err
 }
+
+// ConvertUndefinedErrInto404Err is a method used to parsing errors which related structures undefined in the golangsdk
+// package (so a general method for parsing such errors is needed) and try to convert it to 404 error according to the
+// right error number and code (omitted means all passed).
+// Unsupported status codes (may not enter the convert logic and returned directly):
+//   - 400, 401, 402, 403, 404, 405, 408, 429, 500, 503
+//
+// Arguments:
+//   - err: The error response obtained through HTTP/HTTPS request.
+//   - errStatusNum: The status number of the error, e.g. 409, 415 (The common status numbers are defined in the
+//     golangsdk package, e.g. 400, 403, 500)
+//   - errCodeKey: The key name of the error code in the error response body, e.g. 'error_code', 'err_code'.
+//   - specErrCodes: One or more error codes that you wish to match against the current error, e.g. 'CSS.0004'.
+//
+// Notes: If you missing specErrCodes input, this function will convert all errors which matched your specifies status
+// into 404 errors.
+//
+// How to use it:
+//   - utils.ConvertUndefinedErrInto404Err(err, 415, "error_code")
+//   - utils.ConvertUndefinedErrInto404Err(err, 415, "error_code", "CSS.0004")
+//   - utils.ConvertUndefinedErrInto404Err(err, 415, "error_code", []string{"CSS.0004", "CSS.0005"}...)
+//   - utils.ConvertUndefinedErrInto404Err(err, 415, "")
+//
+// The corresponding processing of this method are as follows:
+//   - The status code has a corresponding structure definition in golangsdk: the error is recorded in the log and the
+//     original error is returned directly.
+//   - The status code (key) not found: return error (404 error: if the errCodeKey is omitted).
+//   - The input of the expected error code(s) is omitted: return 404 error.
+//   - The expected error code(s) matched: return 404 error.
+//   - Other situations: return original error.
+func ConvertUndefinedErrInto404Err(err error, errStatusNum int, errCodeKey string, specErrCodes ...string) error {
```

Comment:
It seems that there is no need to distinguish whether the error code is defined or not.
It is recommended to change the name to "ConvertErrInto404Err" and then remove the check of `definedStatusNumbers`.